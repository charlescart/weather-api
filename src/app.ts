/* eslint-disable no-console */
import app from './server';

const port = process.env.PORT || 3000;

app.listen(port, () => {
  console.log(`Run in the port ${port}`);
});
