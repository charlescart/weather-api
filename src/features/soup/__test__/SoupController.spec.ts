import app from '../../../server';
import request from 'supertest'

describe('SoupController', () => {
  describe('Sopa de letras irregular y multiples palabras a buscar', () => {
    it('POST /api/soup', done => {
      const searchWords: string[] = ["oiE", "ixE", "Ix", "i", "O", "awds", "e4", "", "oi", "eX"];
      const soup: string[][] = [
        ["o", "I", "E"],
        ["I", "i", "X", "4"],
        ["E", "X", "E"],
        ["E", "X"],
        ["E"],
        [""]
      ];

      request(app).post('/api/soup')
        .send({ soup, searchWords })
        .set('Content-Type', 'application/json')
        .set('Accept', 'application/json')
        .expect(200)
        .end((err, res) => {
          if (err) throw err;
          expect(res.body).toMatchSnapshot();
          done();
        });
    });
  });

  describe('matriz 3x3 palabra a buscar ["OIE"]', () => {
    it('POST /api/soup', done => {
      const searchWords: string[] = ["oiE"];
      const soup: string[][] = [
        ["O", "I", "E"],
        ["I", "I", "X"],
        ["E", "X", "E"],
      ];

      request(app).post('/api/soup')
        .send({ soup, searchWords })
        .set('Content-Type', 'application/json')
        .set('Accept', 'application/json')
        .expect(200)
        .end((err, res) => {
          if (err) throw err;
          expect(res.body).toMatchSnapshot();
          done();
        });
    });
  });

  describe('matriz 1x10 palabra a buscar ["OIE"]', () => {
    it('POST /api/soup', done => {
      const searchWords: string[] = ["oiE"];
      const soup: string[][] = [
        ["E", "I", "O", "I", "E", "I", "O", "E", "I", "O"],
      ];

      request(app).post('/api/soup')
        .send({ soup, searchWords })
        .set('Content-Type', 'application/json')
        .set('Accept', 'application/json')
        .expect(200)
        .end((err, res) => {
          if (err) throw err;
          expect(res.body).toMatchSnapshot();
          done();
        });
    });
  });

  describe('matriz 5x5 palabra a buscar ["OIE"]', () => {
    it('POST /api/soup', done => {
      const searchWords: string[] = ["oiE"];
      const soup: string[][] = [
        ["E", "A", "E", "A", "E"],
        ["A", "I", "I", "I", "A"],
        ["E", "I", "O", "I", "E"],
        ["A", "I", "I", "I", "A"],
        ["E", "A", "E", "A", "E"],
      ];

      request(app).post('/api/soup')
        .send({ soup, searchWords })
        .set('Content-Type', 'application/json')
        .set('Accept', 'application/json')
        .expect(200)
        .end((err, res) => {
          if (err) throw err;
          expect(res.body).toMatchSnapshot();
          done();
        });
    });
  });

  describe('matriz 7x2 palabra a buscar ["OIE"]', () => {
    it('POST /api/soup', done => {
      const searchWords: string[] = ["oiE"];
      const soup: string[][] = [
        ["O", "X"],
        ["I", "O"],
        ["E", "X"],
        ["I", "I"],
        ["O", "X"],
        ["I", "E"],
        ["E", "X"],
      ];

      request(app).post('/api/soup')
        .send({ soup, searchWords })
        .set('Content-Type', 'application/json')
        .set('Accept', 'application/json')
        .expect(200)
        .end((err, res) => {
          if (err) throw err;
          expect(res.body).toMatchSnapshot();
          done();
        });
    });
  });

  describe('matriz 1x1 palabra a buscar ["OIE"]', () => {
    it('POST /api/soup', done => {
      const searchWords: string[] = ["oiE"];
      const soup: string[][] = [["E"],];

      request(app).post('/api/soup')
        .send({ soup, searchWords })
        .set('Content-Type', 'application/json')
        .set('Accept', 'application/json')
        .expect(200)
        .end((err, res) => {
          if (err) throw err;
          expect(res.body).toMatchSnapshot();
          done();
        });
    });
  });
});