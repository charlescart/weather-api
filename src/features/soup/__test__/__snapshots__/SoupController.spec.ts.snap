// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoupController Sopa de letras irregular y multiples palabras a buscar POST /api/soup 1`] = `
Object {
  "AWDS": 0,
  "E4": 2,
  "EX": 9,
  "I": 3,
  "IX": 4,
  "IXE": 0,
  "O": 1,
  "OI": 3,
  "OIE": 3,
}
`;

exports[`SoupController matriz 1x1 palabra a buscar ["OIE"] POST /api/soup 1`] = `
Object {
  "OIE": 0,
}
`;

exports[`SoupController matriz 1x10 palabra a buscar ["OIE"] POST /api/soup 1`] = `
Object {
  "OIE": 4,
}
`;

exports[`SoupController matriz 3x3 palabra a buscar ["OIE"] POST /api/soup 1`] = `
Object {
  "OIE": 3,
}
`;

exports[`SoupController matriz 5x5 palabra a buscar ["OIE"] POST /api/soup 1`] = `
Object {
  "OIE": 8,
}
`;

exports[`SoupController matriz 7x2 palabra a buscar ["OIE"] POST /api/soup 1`] = `
Object {
  "OIE": 3,
}
`;
